checkrebuild

```diff
@@ -4,51 +4,168 @@
 
 shopt -s nullglob
 shopt -s extglob
+[ -t 1 ] && declare _terminal_output=1
 
 log="$(mktemp -d)"
+touch "$log/resolved_symbols"
 trap 'rm -rf "$log"' EXIT
 
 verbose=0
-while getopts "v" OPTION; do
+while getopts "vs" OPTION; do
     case $OPTION in
-        v) verbose=1 ;;
+        s) system=1 ;;
+        v) ((verbose+=1)) ;;
         *) exit 1 ;;
     esac
 done
 
 get_unofficial_pkgs() {
-    official_repos="testing core extra community-testing community multilib-testing multilib"
-    comm -23 <(pacman -Qq | sort) <(pacman -Sl $official_repos 2>/dev/null | awk '/\[.*[[:alpha:]]+]$/ {print $2};/\[.*[[:digit:]]+]$/ {print $2}' | sort)
+    if ((system))
+    then pacman -Qq
+    else
+        local_enabled_repos="$(pacconf --repo-list|xargs -I{} bash -c "pacconf --repo={} Server|grep -q -v "file://" && echo {}"|paste -sd ' ' -)"
+        comm -23 <(pacman -Qq | sort) <(eval pacman -Sql "$local_enabled_repos" 2>/dev/null | sort)
```

Comment:
do we really need `eval`s in the code, like here for example? We should review all usages of eval

Reply:
`eval` can be dropped here but we need to remove quotation in `pacman` call, to allow words splitting.